### PR TITLE
Add Release Page

### LINKED ISSUE #44: Add releases
We should find a way to provide releases and/or release notes on our website.

We can probably utilise the GitHub API to get the name of the release as well as a link to the release notes.

This could help people to find them without being "GitHub aware".

### PR DESCRIPTION
**Pull request by DerMolly**
### Description
This PR adds the releases as entries.

### Steps

- [x] Added changes
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc.github.io/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### Related Issue(s)
fixes #44

### CODE REVIEW

**Comment by mrdrogdrog** [comment]
This pr is not ready to review! There are complains by the reuse bot and the header nav link isn't working

**Comment by mrdrogdrog** [comment]
Everything should be fine now

**Comment by SISheogorath** [comment]
Just one question: Why do we duplicate the release notes? We already maintain them in 3 places:

- GitHub releases
- git tags
- release-notes.md in the shipped sources

Now we add the homepage as well?

**Comment by mrdrogdrog** [comment]
We fetch them from github. So it won't be a duplicate.

File: github-releases.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    size: number,
+    browser_download_url: string
+  }[]
+  created_at: string
+  prerelease: boolean
+  draft: boolean
+}
+
+export const githubReleases = async () => {
+  fetch('https://api.github.com/repos/hedgedoc/hedgedoc/releases')
```
Comment (ErikMichelson) [inline github-releases.ts:29]: This `fetch` doesn't have a `.catch(err => ...)`.

**Review by ErikMichelson — CHANGES_REQUESTED** [review]
image
Some more padding on the right of the icon would be nice.

Another thing: You committed all the releases-documents into the repository. As they're rebuild everytime you run `yarn install`, why do we need them in the Git-repo? We could just add an .gitignore entry instead.

**Comment by mrdrogdrog** [comment]
They won't be rewritten every time. Only not existing files will be created.

**Comment by mrdrogdrog** [comment]
We can remove some of them, but e.g. the content of 1.7.0 needs post production.

**Comment by ErikMichelson** [comment]
What kind of post-production is needed for 1.7.0?

**Comment by ErikMichelson** [comment]
But if they aren't rewritten, we should at least compare their content with the API result to respect typo-fixes or other fixes in the release-notes without the need for manual work.

**Comment by DerMolly** [comment]
1.7.0 had `:tada:` which we needed to replace with 🎉.

We just check if the file already exists and only write it, if it doesn't.

So the files will be created automatically if the aren't there already. We could of course only put those files with actual fixes in the repo.

File: layouts/releases/single.html [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                    <div class="row">
+                        <h3 class="mb-3">Docker</h3>
+                    </div>
+                    <div class="row">
+                        <p>
+                            Check out our docker repository on&nbsp;<a
+                                href="https://quay.io/repository/hedgedoc/hedgedoc?tab=tags" target="_blank"
+                                rel="noopener noreferer">quay.io</a>
+                        </p>
+                    </div>
```
[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply layouts/releases/single.html:101]: In the first version I had this, but then we would have to check if the image exists.

[the remark this answers is not in the document]
Comment (SISheogorath) [reply layouts/releases/single.html:101]: I'm for the former. It makes it easier for everyone and doesn't hurt.

**Comment by SISheogorath** [comment]
Is there maybe a Hugo plugin for gihub emojis? I know Jekyll has those in the default GitHub pages setup.

File: github-releases.ts (hunk 1 above)
Comment (DerMolly) [reply github-releases.ts:29]: fixed

**Comment by DerMolly** [comment]
Maybe, but it's still a good idea to make it possible to have changes for the website if we need them. But we probably should remove all files without changes.

**Comment by ErikMichelson** [comment]
Do we want to add the releases folder (with exception for index, and maybe 1.7.0) to gitignore?

Besides that, enabling emoji-shortcode support in hugo is a simple config setting ([enableEmoji](https://gohugo.io/getting-started/configuration/#all-configuration-settings)).

**Comment by mrdrogdrog** [comment]
You can add a directory to the gitignore and override this with `git add -f`

**Comment by mrdrogdrog** [comment]
but yes.. +1 for gitignore

**Comment by DerMolly** [comment]
I just added the `enableEmoji` option, but it still needs some manual tweaking because of newlines:
image
image

**Comment by DerMolly** [comment]
we could just change this in the release. What do you think, @ErikMichelson @SISheogorath ?

**Comment by mrdrogdrog** [comment]
Tbh.. i think this is okay

File: layouts/releases/single.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (DerMolly) [reply layouts/releases/single.html:101]: Maybe we should just show this on the latest release then?

[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply layouts/releases/single.html:101]: @DerMolly and I are unsure how to realize this on an automatic base. We would merge this PR first and then change the docker stuff later.

**Comment by mrdrogdrog** [comment]
Fixed icon width

**Review by mrdrogdrog — APPROVED** [review]

**Review by SISheogorath — APPROVED** [review]

File: layouts/releases/single.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (ErikMichelson) [reply layouts/releases/single.html:101]: Okay, I'll write an issue for that.

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**